processor/resourcedetectionprocessor/internal/kubeadm/config.go

```diff
@@ -0,0 +1,34 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package kubeadm // import "github.com/open-telemetry/opentelemetry-collector-contrib/processor/resourcedetectionprocessor/internal/kubeadm"
+
+import (
+	"github.com/open-telemetry/opentelemetry-collector-contrib/internal/k8sconfig"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/processor/resourcedetectionprocessor/internal/kubeadm/internal/metadata"
+)
+
+type Config struct {
+	k8sconfig.APIConfig `mapstructure:",squash"`
+	ResourceAttributes  metadata.ResourceAttributesConfig `mapstructure:"resource_attributes"`
+	configMapName       string
+	configMapNamespace  string
+}
+
+const defaultConfigMapName = "kubeadm-config"
+const defaultConfigMapNamespace = "kube-system"
+
+// UpdateDefaults validates and update the default config with user's provided settings
+func (c *Config) UpdateDefaults() error {
```

Comment:
The usual way of handling setting default values is in `CreateDefaultConfig()`, is there a reason this functionality can't be handled there?

Reply:
if we potentially want to extend the functionality of the kubeadm detector to read the configmap name + namespace from env variables, we can easily just adapt this function to fetch the env variable and update defaults. But I agree, this is not necessary in the current detector, so can be simplified if needed.

Reply:
I'd prefer to contain as much "setting default values" logic in the `CreateDefaultConfig` as possible for now. We can discuss again in the future if (or when) that case pops up 👍